spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -628,4 +613,50 @@ private Table table() {
       return tableBroadcast.value();
     }
   }
+
+  static class PartitionFieldSortFunction implements SerializableFunction<DataFile, String> {
```

Comment:
You left a comment before about including the function as a static class in here, this might be the trick to get it to work for me.